server/parser/goalParser.ts

```diff
@@ -16,11 +16,11 @@ class GoalParser extends DatabaseParser {
         return this.parseDatabase(query);
     }
 
-    async storeGoal(name : string, description : string, isComplete : boolean, moduleID : number) {
+    async storeGoal(name : string, description : string, goalType: string, isComplete : boolean, moduleID : number, due_date? : string) {
```

Comment:
Instead of passing 6 parameters into this function definition, you can create a new `Goal` type and then pass that in as an object. 

```
type Goal = {
  name: string,
  description: string,
  isComplete: boolean,
  goalType: string,
  moduleId: number,
  dueDate: Date,
}
```

The you can use it like this `async storeGoal(goal: Goal)` and destructure it when you need the values
`const {name, description, etc} = goal`

One more thing, you should pick one naming convention and be consistent in the code. You are using camel case and snake case here

Reply:
I noticed in your types in the client that GoalType is defined as `export type GoalType = "todo" | "daily";`. Since you only allow it to be `"todo" | "daily"`, you should do the same on the backend in your types

```
type GoalType = "todo" | "daily"

type Goal = {
  name: string,
  description: string,
  isComplete: boolean,
  goalType: GoalType,
  moduleId: number,
  dueDate: Date,
}
``